django_informixdb/base.py

```diff
@@ -331,13 +351,25 @@ def _start_transaction_under_autocommit(self):
         self.cursor().execute(start_sql)
 
     def is_usable(self):
+        # We create a cursor and then explicitly close it as there is a bug
+        # that is encountered when relying on garbage collection to close the
+        # cursor: https://github.com/mkleehammer/pyodbc/issues/585
         try:
-            # Use a cursor directly, bypassing Django's utilities.
-            self.connection.cursor().execute("SELECT 1")
+            cursor = self.connection.cursor()
         except pyodbc.Error:
             return False
-        else:
-            return True
+
+        try:
+            cursor.execute(self._validation_query)
+        except pyodbc.Error:
+            return False
+        finally:
+            try:
+                cursor.close()
```

Comment:
I wonder whether it's worth mentioning in a comment *why* the code looks like this, to help a future dev avoid the pitfall of destructors not raising errors properly?

Reply:
There is a comment explaining this at the top of the function 😄

Reply:
Added an extra comment because this whole `return` from within a `finally` is pretty weird.

Reply:
> There is a comment explaining this at the top of the function 😄

Oops 😅